pystac/catalog.py

```diff
@@ -946,7 +953,16 @@ def full_copy(
 
     @classmethod
     def from_file(cls, href: str, stac_io: Optional[pystac.StacIO] = None) -> "Catalog":
+        if stac_io is None:
+            stac_io = pystac.StacIO.default()
+
         result = super().from_file(href, stac_io)
         if not isinstance(result, Catalog):
             raise pystac.STACTypeError(f"{result} is not a {Catalog}.")
+        result._stac_io = stac_io
+
         return result
+
+    @classmethod
+    def identify_dict(cls, d: Dict[str, Any]) -> bool:
```

Comment:
This is defined as an abstract method in `STACObject` so it would need to have a consistent naming convention across all sub-classes. I didn't really like `identify_dict` either, but I couldn't think of a better generic option. Maybe `validate_dict_type`?

Reply:
Although, now that I look at it, it doesn't need to be an abstract class because it's only used in specific sub-class method definitions. I'll update the name as you suggested.

Reply:
Ah, now I remember why I did it this way. I wanted to standardize the name so that a sub-class like `pystac_client.API` could overwrite that method and have the name still make sense.

Something like:

```python
class API(pystac.Catalog):
    def identify_dict(cls, d):
        return super().identify_dict() and "conformance" in d
```

maybe `dict_matches_object_type` would be better.

Reply:
Makes sense - perhaps `matches_object_type` or `is_object_type`, as the `dict` is in the method signature?

Reply:
Renamed in e15c7c04d5904f144050285cf8d453fe8ea701af